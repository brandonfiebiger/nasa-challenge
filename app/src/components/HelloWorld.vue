<template>
  <div class="hello">
    <h1>Nasa Pics!</h1>
    <img v-bind:src="message"/>
  </div>
</template>

<script>
  import { apiKey } from '../../keys/keys.js'
  
export default {
  name: 'HelloWorld',
  props: {
    msg: String
  },
  data() {
    return {
      message: ''
    }
  },
  created: function() {
    fetch(`https://api.nasa.gov/planetary/apod?api_key=${apiKey}`)
      .then(response => response.json())
      .then(image => this.message = image.url)
      .catch(error => console.log(error))
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
