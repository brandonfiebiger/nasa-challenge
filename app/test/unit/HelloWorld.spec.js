import { mount } from '@vue/test-utils';
import HelloWorld from '../../src/components/HelloWorld.vue';

describe('HelloWorld', () => {

  beforeEach(() => {
    window.fetch = jest.fn()
  })
  it('should exist', () => {
    const wrapper = mount(HelloWorld);
    expect(wrapper).toMatchSnapshot()
  })
})