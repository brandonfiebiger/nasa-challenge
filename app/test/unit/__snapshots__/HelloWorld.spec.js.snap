// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld should exist 1`] = `
<div class="hello">
  <h1>Nasa Pics!</h1> <img src=""> <button>Today's Photo</button> <button>Yesterday's Photo</button>
</div>
`;

exports[`HelloWorld should match the snapshot 1`] = `
<div class="hello">
  <h1>Nasa Pics!</h1> <img src=""> <button>Today's Photo</button> <button>Yesterday's Photo</button>
</div>
`;
